### PR TITLE
[cmake] consolidate set_target_properties() calls

### PR DESCRIPTION
**Pull request by jameslamb**
Another step towards simplifying this project's CMake configuration (are you sensing a theme in my recent PRs? 😂 ).

CMake's `set_target_properties()` accepts multiple targets and multiple properties ([CMake docs](https://cmake.org/cmake/help/latest/command/set_target_properties.html)).

This proposes taking advantage of that to consolidate a few calls.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -655,20 +663,22 @@ if(__INTEGRATE_OPENCL)
...
   if(BUILD_CLI)
-    set_target_properties(lightgbm PROPERTIES CUDA_RESOLVE_DEVICE_SYMBOLS ON)
+    set_target_properties(
+      lightgbm
+      PROPERTIES
+        CUDA_ARCHITECTURES ${CUDA_ARCHS}
+        CUDA_RESOLVE_DEVICE_SYMBOLS ON
+    )
   endif()
-  set_target_properties(_lightgbm PROPERTIES CUDA_RESOLVE_DEVICE_SYMBOLS ON)
```
Comment (StrikerRUS) [inline CMakeLists.txt:671]: This line was lost in new config.

**Review by StrikerRUS — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (jameslamb) [reply CMakeLists.txt:671]: ah! Too many similarly-named properties 😂 

Thank you for catching this, I've fixed that in recent commits.

File: CMakeLists.txt [hunk 2]
```diff
@@ -655,20 +663,22 @@ if(__INTEGRATE_OPENCL)
...
-  if(BUILD_CLI)
-    set_target_properties(lightgbm PROPERTIES CUDA_ARCHITECTURES ${CUDA_ARCHS})
-  endif()
 
-  set_target_properties(lightgbm_objs PROPERTIES CUDA_SEPARABLE_COMPILATION ON)
+  set_target_properties(
+    lightgbm_objs _lightgbm
+    PROPERTIES
+      CUDA_ARCHITECTURES ${CUDA_ARCHS}
+      CUDA_SEPARABLE_COMPILATION ON
```
[the remark this answers is not in the document]
Comment (jameslamb) [reply CMakeLists.txt:671]: ah! Thank you for catching that, I've updated it in recent commits.

This also led me to question what this property actually does. Here's a simple explanation of it: https://stackoverflow.com/a/50557765/3986677

**Review by StrikerRUS — APPROVED** [review]
Thanks for solving this CMake puzzle! 😄

**Comment by jameslamb** [comment]
Sure! I've been learning a lot about CMake lately, hoping to continue simplifying and improving LightGBM's CMake stuff.

I'm especially interested in making LightGBM more friendly to build and link to for repackagers like @barracuda156 (see for example #6489).

**Pull request merged**